Source/AssetRipper.Processing/AnimatorControllers/AnimatorStateMachineContext.cs

```diff
@@ -0,0 +1,759 @@
+						break;
+					}
+				}
+			}
+			if (parentStateMachine == null)
+			{
+				// ensure parent for stateMachine
+				parentStateMachine = RootStateMachine;
+			}
+			// ensure all StateMachines have Name
+			if (childStateMachine.Name.IsEmpty) // || childStateMachine.Name == FlagName
+			{
+				// for recompiling the AnimatorController back to Release and getting the original StateMachine path hashes:
+				//   stateMachine_FullPath = Crc32Algorithm.ReverseAscii(StateMachineFullPathIDs[childIndex], Parent_StateMachine_FullPath_Name + ".");
+				//   stateMachine.Name = stateMachine_FullPath[(Parent_StateMachine_FullPath_Name.Length+1)..];
+				// may need to check and set parent StateMachine Name recursively, if its still Empty or FlagName
+				childStateMachine.Name = $"Empty_{StateMachineFullPathIDs[childIndex]}";
+			}
+
+			// set Child StateMachine for its found Parent
+			if (parentStateMachine.Has_ChildStateMachines())
+			{
+				ChildAnimatorStateMachine child = parentStateMachine.ChildStateMachines.AddNew();
+				child.StateMachine.SetAsset(parentStateMachine.Collection, childStateMachine);
+			}
+			else
+			{
+				PPtr_AnimatorStateMachine_4 child = parentStateMachine.ChildStateMachine.AddNew();
+				child.SetAsset(parentStateMachine.Collection, childStateMachine);
+			}
+
+			// add StateMachine Transitions (Transitions from Child are stored in Parent)
+			SelectorTransitionConstant[]? exitTransitions = ExitTransitions[childIndex];
+			if (exitTransitions != null)
+			{
+				// check if its the default exit transition, to not add it
+				if (exitTransitions.Length == 1 && exitTransitions[0].ConditionConstantArray.Count == 0)
+				{
+					uint stateDestinationId = exitTransitions[0].Destination;
+					if (stateDestinationId < StateMachineTransitionFlag)
+					{
+						int stateDestinationIndex = (int)stateDestinationId;
+						int defaultStateIndex = StateContext.GetStateIndex(parentStateMachine.DefaultStateP);
+						if (defaultStateIndex == stateDestinationIndex)
+						{
+							continue;
+						}
+					}
+				}
+
+				AssetPair<PPtr_AnimatorStateMachine_5, AssetList<PPtr_AnimatorTransition>> newPair =
+						parentStateMachine.StateMachineTransitions!.AddNew();
+				newPair.Key.SetAsset(parentStateMachine.Collection, childStateMachine);
+				PPtrAccessList<PPtr_AnimatorTransition, IAnimatorTransition> transitions = new(newPair.Value, parentStateMachine.Collection);
+				for (int j = 0; j < exitTransitions.Length; j++)
+				{
+					SelectorTransitionConstant selectorTransition = exitTransitions[j];
+					IAnimatorTransition? transition = CreateAnimatorTransition(childIndex, false, selectorTransition);
+					if (transition != null)
+					{
+						transitions.Add(transition);
+					}
+				}
+
+				if (transitions.Count != 0)
+				{
+					newPair.Value.Capacity = newPair.Value.Count;
+				}
+				else
+				{
+					parentStateMachine.StateMachineTransitions.RemoveAt(parentStateMachine.StateMachineTransitions.Count - 1);
```

Comment:
If you add a `AccessListBase<>::RemoveAt(System.Index)` overload, you should be able to do `RemoveAt(^1)` here.